### PR TITLE
Granular linting

### PR DESCRIPTION
**Pull request by NiklasBuchfink**
This PR adds granular linting to the SDK to reduce performance issues and enable synchronized updates.

- [x] Fix executing loadProject multiple times within Fink
- [x] Create reactiveMap for messageLintQuery
- [x] Remove debounce
- [x] Handle adding new messages for Paraglide JS and VS code extension 
- [x] Non-blocking lint calculation / lazy-linting by debouncing
- [x] Update & pass all tests
- [x] Make sure other apps still work

### CODE REVIEW

File: inlang/source-code/editor/src/pages/@host/@owner/@repository/State.tsx [hunk 1]
```diff
@@ -208,9 +208,6 @@ export function EditorStateProvider(props: { children: JSXElement }) {
 	// get lix errors
 	const [lixErrors, setLixErrors] = createSignal<ReturnType<Repository["errors"]>>([])
 	createEffect(() => {
-		// reset errors to empty on repo changes
-		setLixErrors([])
-
 		repo()?.errors.subscribe((errors) => {
 			setLixErrors(errors)
 		})
```
Comment (samuelstroschein) [inline inlang/source-code/editor/src/pages/@host/@owner/@repository/State.tsx:213]: @NiklasBuchfink why can't you directly subscropbe to `repo().errors`? 

Does no wrapper for repo reactivity exist?

Comment (NiklasBuchfink) [reply inlang/source-code/editor/src/pages/@host/@owner/@repository/State.tsx]: You are right, moved it behind the repo resource and removed the effect and reset. Everything works, including the resets after changing the branch 👍

**Comment by NiklasBuchfink** [comment]
@samuelstroschein The last difficult step is now the non-blocking linting calls. This will save up to one minute of startup time with extremely large repos (in terms of translations).

**Comment by samuelstroschein** [comment]
@NiklasBuchfink we will look into this as part of #455

**Comment by NiklasBuchfink** [comment]
> @NiklasBuchfink we will look into this as part of #1772

Then I have to debounce the write file again because updating a single message with more than 20 languages and four lint rules has already become a performance problem.

The route course of #1772:
Reactivity with SolidJS is granular; it just needs to be engineered that way.
The file watcher is **not** granular and therefore stupidly recreates everything instead of updating the change diff here and there in the reactivity system.

File: inlang/source-code/sdk/src/createMessageLintReportsQuery.ts [hunk 2]
```diff
@@ -1,50 +1,78 @@
...
-				}).then((report) => {
-					if (
-						report.errors.length === 0 &&
-						JSON.stringify(index.get(message.id)) !== JSON.stringify(report.data)
-					) {
-						index.set(message.id, report.data || [])
+	createEffect(() => {
+		if (rulesArray) {
+			for (const messageId of messagesQuery.includedMessageIds()) {
+				messagesQuery.get.subscribe({ where: { id: messageId } }, async (message) => {
```
Comment (samuelstroschein) [inline inlang/source-code/sdk/src/createMessageLintReportsQuery.ts:47]: why do you use subscribe? 

you can directly use `get()` without the subscribe callback.

File: inlang/source-code/sdk/src/createMessageLintReportsQuery.ts [hunk 3]
```diff
@@ -1,50 +1,78 @@
...
+						await lintSingleMessage({
+							rules: rulesArray,
+							settings: settingsObject(),
+							messages: messages,
+							message: message,
+						}).then((report) => {
+							if (report.errors.length === 0 && index.get(messageId) !== report.data) {
+								index.set(messageId, report.data)
+							}
+						})
```
Comment (samuelstroschein) [inline inlang/source-code/sdk/src/createMessageLintReportsQuery.ts:58]: Why are you awaiting and using a `.then` callback? The `await` keyword is probably ignored. Decide for one. Given that you do not want to run lints when the project loads, `.then` seems to be the correct choice.

File: inlang/source-code/sdk/src/createMessageLintReportsQuery.ts [hunk 4]
```diff
@@ -1,50 +1,78 @@
...
+									messages: messages,
+									message: message,
+								}).then((report) => {
+									if (report.errors.length === 0 && index.get(messageId) !== report.data) {
+										index.set(messageId, report.data)
+									}
+								})
+							},
+							{ atBegin: false }
+						)(message)
```
Comment (samuelstroschein) [inline inlang/source-code/sdk/src/createMessageLintReportsQuery.ts:75]: 1. Why a debouncer here?
2. Async then problem as above.

File: inlang/source-code/sdk/src/createMessageLintReportsQuery.ts (hunk 2 above)
Comment (NilsJacobsen) [reply inlang/source-code/sdk/src/createMessageLintReportsQuery.ts:47]: image

I don't think so. It's not being converted to a signal. It only returns a value this way.

File: inlang/source-code/sdk/src/createMessageLintReportsQuery.ts (hunk 4 above)
Comment (NiklasBuchfink) [reply inlang/source-code/sdk/src/createMessageLintReportsQuery.ts:75]: 1. Debouncing is required until this is implemented https://github.com/inlang/monorepo/issues/1817
**Background:** Linting is triggered on every key change; no longer on the debounced save. This works for projects with only a few languages and lint rules, but for projects with 20+ languages and 4 lint rules, it blocks the main thread and the pattern editor field lags massively behind (loss of real-time feedback).

2. Agreed, I'll remove it

File: inlang/source-code/sdk/src/createMessageLintReportsQuery.ts (hunk 2 above)
Comment (NilsJacobsen) [reply inlang/source-code/sdk/src/createMessageLintReportsQuery.ts]: haha sdk trapped me :D

Comment (NiklasBuchfink) [reply inlang/source-code/sdk/src/createMessageLintReportsQuery.ts]: @samuelstroschein We changed this. It is now a signal within a createEffect.

File: inlang/source-code/sdk/src/createMessageLintReportsQuery.ts [hunk 5]
```diff
@@ -44,7 +44,8 @@ export function createMessageLintReportsQuery(
 	createEffect(() => {
 		if (rulesArray) {
 			for (const messageId of messagesQuery.includedMessageIds()) {
-				messagesQuery.get.subscribe({ where: { id: messageId } }, (message) => {
+				createEffect(() => {
+					const message = messagesQuery.get({ where: { id: messageId } })
```
[the remark this answers is not in the document]
Comment (NiklasBuchfink) [reply inlang/source-code/sdk/src/createMessageLintReportsQuery.ts:48]: Its ugly but:
Effect 1. is triggered when a new message is created and all reports are created again (requirement for the VS code extension)
Effect 2. is triggered when a message is updated and a single new report is created.

**Pull request merged**